protostar/testing/environments/test_execution_environment.py

```diff
@@ -34,6 +34,9 @@ def __init__(self, state: TestExecutionState):
         self._finish_hook = Hook()
 
     async def execute(self, function_name: str) -> TestExecutionResult:
+        assert (
```

Comment:
If TestExecutionState required contract then we could avoid this assert

Reply:
classes restructured